crates/hir/src/code_model.rs

```diff
@@ -1098,7 +1100,24 @@ impl_from!(
 );
 
 impl GenericDef {
-    pub fn params(self, db: &dyn HirDatabase) -> Vec<TypeParam> {
+    pub fn params(self, db: &dyn HirDatabase) -> Vec<GenericParam> {
+        let generics: Arc<hir_def::generics::GenericParams> = db.generic_params(self.into());
```

Comment:
Seems like this type signature can be removed?

Reply:
The turbofish assist would use it. Though I'm not sure if that should even care about lifetimes, probably not? https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/assists/src/handlers/add_turbo_fish.rs#L50-L55

Reply:
I mean

```suggestion
        let generics = db.generic_params(self.into());
```